### PR TITLE
[FLOSS H2] Added more key copy tests

### PR DESCRIPTION
**Pull request by muskater**
[FLOSS H2]

For this pull request I have tried to fix a ToDo in the `keyTest.c` file: I have added more test cases for the keyCopy function.
Should I remove the "FIXME?"

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by muskater** [comment]
I have read in the [documentation](https://www.libelektra.org/testing/testing), that if tests use the `system/spec namespaces` they need to be marked as `kdbtests`. Do I have to do this for my new test cases since I use these namespaces? 

I could not find where this property needs to be set.

**Comment by muskater** [comment]
Does anyone know why the `continuous-integration/jenkins/pr-merge` failed? Unfortunately, I don't understand the error message.

**Comment by markus2330** [comment]
> Should I remove the "FIXME?"

Yes, you can remove the FIXME!

>  that if tests use the system/spec namespaces they need to be marked as kdbtests. Do I have to do this for my new test cases since I use these namespaces?

Only if you actually set these keys with kdbSet. As this does not happen, you do not need to mark them with kdbtests. I clarified this in  #4127

> Does anyone know why the continuous-integration/jenkins/pr-merge failed? Unfortunately, I don't understand the error message.

Thank you for reporting, I forwarded it to #160, as it seems to be an infrastructure problem.

Please readd "help needed" if further help is needed.

**Comment by markus2330** [comment]
jenkins build libelektra please

File: tests/ctest/test_key.c [hunk 1]
```diff
@@ -776,8 +775,53 @@ static void test_keyCopy (void)
...
+
+	succeed_if (keyCopy (NULL, keyNorm, KEY_CP_NAME) == NULL, "could copy to NULL");
+	succeed_if (keyCopy (NULL, keyLock, KEY_CP_NAME) == NULL, "could copy to NULL");
+
+	Key * keyBin = keyNew ("user:/binary/foo", KEY_FLAGS, KEY_BINARY, KEY_END);
+
+	succeed_if (keyIsBinary (keyBin), "error creating binary key");
+	succeed_if (keyCopy (keyNorm, keyBin, KEY_CP_STRING) == NULL, "could copy string to binary key");
+	succeed_if (keyCopy (keyNorm, keyBin, KEY_CP_NAME) != NULL, "could not copy name to binary key");
+
```
Comment (markus2330) [inline tests/ctest/test_key.c:813]: there is no check if keyNorm actually contains the wanted content.

Comment (muskater) [reply tests/ctest/test_key.c:813]: This has slipped my mind. Thank you for pointing this out!

**Comment by markus2330** [comment]
There is still a reformatting issue. The rebasing is not done perfectly: there are some commits not by you between the commits.

**Comment by muskater** [comment]
I added some more thorough test cases and fixed the reformatting issue. Also it is now rebased with the master

**Comment by markus2330** [comment]
Great job :sparkling_heart:

**Pull request merged**